src/main/java/com/palantir/refreshable/DefaultRefreshable.java

```diff
@@ -45,20 +47,32 @@
  */
 final class DefaultRefreshable<T> implements SettableRefreshable<T> {
     private static final Logger log = LoggerFactory.getLogger(DefaultRefreshable.class);
-    /**
-     * Every ten times refreshable.map is called without an update we must purge SelfRemovingMapSubscriber instances
-     * whose children have been reaped.
-     */
-    private static final int CLEAN_THRESHOLD = 10;
+    private static final Cleaner REFRESHABLE_CLEANER = Cleaner.create();
```

Comment:
Good catch, we should definitely use a threadfactory here to give the thread a helpful name. Probably not worth adding metrics since there should always be a single thread anchored on this type.

Reply:
The metrics comments reminds me, I forgot to put this here: is there a way to get metrics on the internal queue size?

Reply:
Not that I'm aware of